<!-- Launch file to launch the C++ sccript -->

<launch>

  <arg name="MAV_NAME" default="$(optenv MAV_NAME quadrotor)"/>  <!-- Getting MAV_NAME defined as an environment variable -->
  <arg name="yaml_file" default="$(find monitor_cpp)/config/monitor.yaml" /> <!-- Getting the yaml file -->

  <!-- Calling the C++ script -->
  <node name="monitor_cpp_node" pkg="monitor_cpp" type="monitor_cpp_node" output="screen">
    <param name="mav_name" value="$(arg MAV_NAME)" /> <!-- Passing the mav_name to the C++ script -->
    <param name="yaml_file" value="$(arg yaml_file)" /> <!-- Passing the yaml file to the C++ script -->
  </node>
  
</launch>
